mmf/utils/build.py

```diff
@@ -226,7 +226,24 @@ def build_optimizer(model, config):
             )
 
     parameters = get_optimizer_parameters(model, config)
-    optimizer = optimizer_class(parameters, **params)
+
+    if optimizer_config.get("enable_state_sharding", False):
```

Comment:
It would make sense to write a simple test to test this. We have had issue with `build_optimizer` in past.